### ISSUE TITLE
[glsl-out] Switches don't include break

### ISSUE THREAD

**Issue by PolyMeilex** [gfx-rs/naga#1309]
I have this wgsl shader (part of [wgpu_glyph](https://github.com/hecrj/wgpu_glyph/blob/master/src/shader/glyph.wgsl)):
```rs
switch (i32(input)) {
        case 0: {
            
        }
        case 1: {
            
        }
        case 2: {
            
        }
        case 3: {
           
        }
}
```
Before this change: https://github.com/gfx-rs/naga/commit/02c74b50021379092a9d4567fd11aeb0879c5405#diff-cc33bd3c8c18000d3ee7b6f5578e8fa07ffd08d87b0aa3d9525a107937fb3d3f

it used to output this glsl:
```glsl
switch(int(input)) {
            case 0:
                break;
            case 1:
                break;
            case 2:
                break;
            case 3:
                break;
}
```
Now it outputs:
```glsl
switch(int(input)) {
            case 0:

            case 1:

            case 2:

            case 3:
}
```

This breaks wgpu_glyph on OpenGL backend. ~~I'm not sure if this is correct behavior for wgsl, does it brake by default? Judging by other backend, I always thought so.~~

**Comment by JCapucho** [comment]
This is indeed incorrect, wgsl by default breaks unless a falltrough statement is present at the tail of the case, in which case it will be continue to the next case.

**Comment by PolyMeilex** [comment]
btw, I don't want to sound too whiny, but the fact that this glsl change was buried in commit with hlsl prefix was super confusing, I initially gave up on trying to figure out source of this problem, because I skiped all of the hlsl commits. I'm glad that I came back with diferent mindset and chceked every comit one by one.

tldr: I have nothing valuable to say I just felt like I want to complain a little bit 😅
